checkbox-ng/plainbox/impl/session/system_information.py

```diff
@@ -0,0 +1,241 @@
+        Runs the version_cmd returning its output
+
+        :returns: the version fetched form the version_cmd if it runs
+                  succesfully
+        :returns: a failure message with the exception as a postfix if
+                  version_cmd fails
+        """
+        try:
+            return check_output(
+                self.version_cmd,
+                universal_newlines=True,
+                stderr=STDOUT,
+            )
+        except CalledProcessError as e:
+            return "Failed to collect with error: {}".format(e)
+
+    def collect_outputs(self) -> "(OutputSuccess|OutputFailure)":
+        """
+        Runs the collection_cmd and creates an output.
+
+        :returns: (OutputSuccess, 0) if the command returns 0 and the output
+                  is json parsable
+        :returns: (OutputFailure, N) if the command returns (N != 0) or the
+                  output is not json parsable
+        """
+        collection_result = run(
+            self.collection_cmd,
+            universal_newlines=True,
+            stdout=PIPE,
+            stderr=PIPE,
+        )
+        if collection_result.returncode != 0:
+            outputs = OutputFailure(
+                stdout=collection_result.stdout,
+                stderr=collection_result.stderr,
+                return_code=collection_result.returncode,
+            )
+        else:
+            try:
+                json_out = json.loads(collection_result.stdout)
+                outputs = OutputSuccess(
+                    json_out,
+                    stderr=collection_result.stderr,
+                )
+            except json.JSONDecodeError as e:
+                output = (
+                    "Failed to decode with error: {}\n"
+                    "Collection output:\n{}"
+                ).format(str(e), collection_result.stdout)
+                outputs = OutputFailure(
+                    stdout=output,
+                    stderr=collection_result.stderr,
+                    return_code=0,
+                )
+        return outputs
+
+    def collect(self) -> CollectionOutput:
+        version_str = self.collect_version()
+        outputs = self.collect_outputs()
+
+        return CollectionOutput(tool_version=version_str, outputs=outputs)
+
+
+class InxiCollector(Collector):
+    COLLECTOR_NAME = "inxi"
+
+    def __init__(self):
+        super().__init__(
+            collection_cmd=[
+                str(system_information.INXI_PATH),
+                "--admin",
+                "--tty",
+                "-v8",
+                "--output",
+                "json",
+                "--output-file",
+                "print",
+                "-c0",
+            ],
+            version_cmd=[str(system_information.INXI_PATH), "--vs"],
+        )
+
+
+if __name__ == "__main__":
+    collection = collect()
+    to_dump = {
+        name: collected.to_dict() for (name, collected) in collection.items()
+    }
+    print(json.dumps(to_dump, indent=4))
```

Comment:
I'm guessing this is a debugging leftover :)

Reply:
This is intentional, it is pretty nice to have and not expensive at all! (I think this also came out of a discussion we had before)